### PR TITLE
Create getters for station height and start

### PR DESCRIPTION
**Pull request by Gymnasiast**

### CODE REVIEW

File: src/openrct2/ride/Station.cpp [hunk 1]
```diff
@@ -434,3 +432,19 @@ void ride_set_exit_location(Ride* ride, const int32_t stationIndex, const TileCo
...
+void RideStation::SetBaseZ(int32_t newZ)
+{
+    Height = newZ / 8;
+}
+
+CoordsXYZ RideStation::GetStart() const
+{
+    TileCoordsXYZ stationTileCoords{ Start.x, Start.y, Height };
+    return stationTileCoords.ToCoordsXYZ();
+}
```
Comment (tupaschoal) [inline src/openrct2/ride/Station.cpp]: I'm thinking if it maybe shouldn't return the `TileCoordsXYZ` and we get to call `ToCoordsXYZ` on the oustide.

File: src/openrct2/ride/Ride.cpp [hunk 2]
```diff
@@ -7081,23 +7080,16 @@ void sub_6CB945(Ride* ride)
                     if (!ride_get_exit_location(ride, stationId).isNull())
                         break;
 
-                    ride_set_exit_location(
-                        ride, stationId,
-                        { location.x / 32, location.y / 32, ride->stations[stationId].Height,
-                          (uint8_t)tileElement->GetDirection() });
+                    CoordsXYZD loc = { location, ride->stations[stationId].GetBaseZ(), (uint8_t)tileElement->GetDirection() };
```
Comment (tupaschoal) [inline src/openrct2/ride/Ride.cpp]: While you're at it, do `static_cast<Direction>(tileElement->GetDirection())` instead. I've been trying to get away with these C casts.

File: src/openrct2/ride/Ride.cpp [hunk 3]
```diff
@@ -7081,23 +7080,16 @@ void sub_6CB945(Ride* ride)
...
                         break;
 
-                    TileCoordsXYZD entranceLocation = {
-                        location.x / 32,
-                        location.y / 32,
-                        ride->stations[stationId].Height,
-                        (uint8_t)tileElement->GetDirection(),
-                    };
-                    ride_set_entrance_location(ride, stationId, entranceLocation);
+                    CoordsXYZD loc = { location, ride->stations[stationId].GetBaseZ(), (uint8_t)tileElement->GetDirection() };
```
Comment (tupaschoal) [inline src/openrct2/ride/Ride.cpp]: Same as above.

File: src/openrct2/ride/Station.cpp (hunk 1 above)
Comment (Gymnasiast) [reply src/openrct2/ride/Station.cpp]: I haven't done that because we're moving everything to Coords.

File: src/openrct2/ride/Ride.cpp (hunk 2 above)
Comment (Gymnasiast) [reply src/openrct2/ride/Ride.cpp]: GetDirection() already returns a `Direction`, so casting that to `Direction` is a bit pointless.

Comment (tupaschoal) [reply src/openrct2/ride/Ride.cpp]: Even better, we can drop the casts and be happy :)

**Pull request merged**